pkg/cmd/repo/clone/clone.go

```diff
@@ -157,7 +169,16 @@ func cloneRun(opts *CloneOptions) error {
 
 	gitClient := opts.GitClient
 	ctx := context.Background()
-	cloneDir, err := gitClient.Clone(ctx, canonicalCloneURL, opts.GitArgs)
+
+	var cloneDir string
+
+	for i := 0; i < retryLimit; i++ {
+		cloneDir, err = gitClient.Clone(ctx, canonicalCloneURL, opts.GitArgs)
+		if err == nil {
+			break
+		}
+		time.Sleep(retryBackoff * (time.Duration(i) + 1))
```

Comment:
Testable Go code should never be allowed to sleep the same period of time in tests as it does during normal runtime. Even a relatively short duration like sleeping 100ms would make tests really slow. Finding a way to inject a shorter (fake) Duration in tests would be crucial.

Also, the Sleep statement would not return early if a Context has been cancelled, therefore if a Context is involved, it would be better to use a Timer and use a `select` statement to react to either a Timer completing or a Context being cancelled, whichever comes first. Example: https://github.com/cli/oauth/blob/f564fe4e65e587c8f395989f875d6b2f0c1c8260/device/poller.go#L30-L37

Reply:
This is beyond what I'm up for doing at this time. 

It should involve introducing a `poller.go`, and really there should be a `poller_test.go` -- and even that repo doesn't have one.